src/max_encoded_len.rs

```diff
@@ -34,6 +34,10 @@ pub trait MaxEncodedLen: Encode {
 	fn max_encoded_len() -> usize;
 }
 
+/// Similar to [`MaxEncodedLen`], but will return the maximum encoded length of the compact encoding
+pub trait MaxCompactEncodedLen {
+	fn max_compact_encoded_len() -> usize;
```

Comment:
or maybe we just add it to MaxEncodedLen with a default implementation, so we don't have to introduce an extra trait 🤔

Reply:
actually, it might be cleaner to use a `HasCompactMaxEncodedLen` like we do with `HasCompact` for encoding